redwood-compose/src/commonMain/kotlin/app/cash/redwood/compose/WidgetApplier.kt

```diff
@@ -59,37 +64,102 @@ internal class WidgetApplier<W : Any>(
     onEndChanges.invoke()
   }
 
-  override fun insertTopDown(index: Int, instance: Widget<W>) {
+  override fun insertTopDown(index: Int, instance: Node<W>) {
     check(!closed)
 
-    if (instance is ChildrenWidget) {
-      instance.children = instance.accessor!!.invoke(current)
-      instance.accessor = null
+    if (instance is ChildrenNode) {
+      @Suppress("UNCHECKED_CAST") // Guaranteed by generated code.
+      val widgetNode = current as WidgetNode<Widget<W>, W>
+      instance.attachTo(widgetNode.widget)
     } else {
-      val current = current as ChildrenWidget
-      current.children!!.insert(index, instance)
+      @Suppress("UNCHECKED_CAST") // Guaranteed by generated code.
+      val widgetNode = instance as WidgetNode<Widget<W>, W>
+      val children = (current as ChildrenNode<W>).children
+
+      widgetNode.container = children
+      children.insert(index, widgetNode.widget)
     }
   }
 
-  override fun insertBottomUp(index: Int, instance: Widget<W>) {
+  override fun insertBottomUp(index: Int, instance: Node<W>) {
     // Ignored, we insert top-down.
   }
 
   override fun remove(index: Int, count: Int) {
     check(!closed)
 
-    val current = current as ChildrenWidget
-    current.children!!.remove(index, count)
+    val current = current as ChildrenNode
+    current.children.remove(index, count)
   }
 
   override fun move(from: Int, to: Int, count: Int) {
     check(!closed)
 
-    val current = current as ChildrenWidget
-    current.children!!.move(from, to, count)
+    val current = current as ChildrenNode
+    current.children.move(from, to, count)
   }
 
   override fun onClear() {
     closed = true
   }
 }
+
+/** @suppress For generated code usage only. */
+@RedwoodCodegenApi
+public sealed interface Node<W : Any>
+
+/**
+ * A node which wraps a [Widget] and also holds onto the [Widget.Children] in which the widget
+ * is placed. The generics of this class are a little funky in order to avoid casts of the widget
+ * in the generated composables.
+ *
+ * @suppress For generated code usage only.
```

Comment:
Why `@suppress` ?

Reply:
Hides it from docs. No one should ever touch these things, but they're public because generated composable implementation must.